lib/pure/unicode.nim

```diff
@@ -1489,6 +1502,130 @@ proc capitalize*(s: string): string {.noSideEffect, procvar,
 
   result = $toUpper(rune) & substr(s, i)
 
+proc normalizeUnicode*(s: string): string {.noSideEffect,
+  rtl, extern: "nucNormalize".} =
+  ## Normalizes the string ``s``.
+  ##
+  ## That means to convert it to lower case and remove any ``'_'``.
+  const
+    underscore = '_'.Rune
+  var
+    i = 0
+    written = 0
+    rune: Rune
+  result = newString(len(s))
+  while i < len(s):
+    fastRuneAt(s, i, rune)
+    if rune.isUpper():
+      rune = rune.toLower()
+    if rune != underscore:
+      rune.fastToUTF8Copy(result, written)
+
+proc strip*(s: string, leading = true, trailing = true,
+  runes: openarray[Rune] = spaceRanges): string {.noSideEffect,
+  rtl, extern: "nucStrip".} =
+  var
+    s_i = 0
```

Comment:
You probably just copied most of this code, but any chance we could get more descriptive variable names? :)